core/src/main/java/org/axonframework/eventhandling/saga/SagaLifecycle.java

```diff
@@ -113,7 +114,7 @@ public static void end() {
      * @return the thread's current {@link SagaLifecycle}
      */
     protected static SagaLifecycle getInstance() {
-        SagaLifecycle instance = CURRENT_SAGA_LIFECYCLE.get();
+        SagaLifecycle instance = Scope.getCurrentScope();
         if (instance == null) {
```

Comment:
This will never happen (`instance == null`), as `getCurrentScope` will throw an exception when no Scope has been defined/set.